src/Workspaces/Core/Portable/ReassignedVariable/AbstractReassignedVariableService.cs

```diff
@@ -189,7 +189,7 @@ bool TryGetParameterLocation(IParameterSymbol parameter, out TextSpan location)
                         return true;
                     }
                 }
-                else if (parameter.ContainingSymbol.Name == WellKnownMemberNames.TopLevelStatementsEntryPointMethodName)
+                else if (parameter.ContainingSymbol?.Name == WellKnownMemberNames.TopLevelStatementsEntryPointMethodName)
```

Comment:
I don't think parameters are ever going to have a null containing symbol?

Reply:
If that is the case is it worth adding a new `ISymbol ContainingSymbol` member to `IParameterSymbol` which is non-nullable?

Reply:
> If that is the case is it worth adding a new `ISymbol ContainingSymbol` member to `IParameterSymbol` which is non-nullable?

It might be worth adding that to every symbol type except `IAssemblySymbol`.

Reply:
Same for `INamespaceSymbol` too correct? Or maybe it would work with a `[MemberNotNullWhen]` on `IsGolbalNamespace`

Reply:
> If that is the case is it worth adding a new ISymbol ContainingSymbol member to IParameterSymbol which is non-nullable?

I was about to ask this -- it seems like there's a number of places that'd help in the IDE.

Reply:
(and the places it doesn't immediately help we could also switch over to pattern matching)

Reply:
This is meant to be mostly a "just annotate PR" that is attempting to avoid too much functionality change. I'm mentally weighing whether new members do / don't cross that line here. Happy to push forward with that though if you all think it's worth it or keep the PR mostly where it is and file a follow up issue to do the new members.

Reply:
I think I'd rather have the refactor in a separate PR, but I won't be opposed if you want to do it in this one instead. If you do, I'd just ask that when merging, keeping that as a separate commit.

Reply:
I also look forward to that PR teaching me about where my assumptions about `ContainingSymbol` are incorrect. I'm _pretty_ sure that only `IMethodSymbol`s and `IAssemblySymbol`'s can have null `ContainingSymbol`s, but an intensive effort to add the annotations in `new` properties might reveal that to be faulty.

Reply:
The refactor is going to change public API surface. I did a quick spike to see how this would work here 

https://github.com/jaredpar/roslyn/commit/1d6b9464eff492184c8b4ce0b6483936f83fd604

It does work quite nice but this is new public API so not sure how you feel about that. Basically does it need to go through public API review?

Reply:
That is the other reason I'm leaning towards doing the refactor in a separate PR. A lot of the underyling symbol types aren't null annotated so we're somewhat guessing which types do / don't return a non-null `ISymbol`

Reply:
Thought about this overnight and I think we need to do the API changes in a separate PR. The reasoning is:

1. That change is going to need to go through API review.
2. That change requires a lot more work to be correct. I think to really understand what can / can't be `null` at that level we need to annotate a lot of the concrete symbols and that is a decently sized under taking.

The goal for this change was to get `ISymbol` into a better state such that our analyzer / generator customers will make less mistakes here. Moving this API, and the others in the follow up PRs, to be properly `ISymbol?` will achieve that so I think there is value in this change on its own. 

Any objections with splitting this up like that?

Reply:
No objections from me.

Reply:
No objections from me as far as splitting up work, but @jaredpar is there something to track the work, like a bug? The bug would be helpful since you could TODO the various places you added a check where we could do something better instead.

Reply:
And for **this immediate change**, should it be a ! instead of a ?, since we don't think it can be null then? That's similar to what you did in other places in this PR.

Reply:
@jaredpar My worry about adding the non-nullable `new` members in a separate PR is that this PR will then add a lot of unnecessary assertions and null checks that aren't actually necessary, because of warnings that wouldn't be there if the new non-nullable members were added at the same time.